extensions/kubeapi/rbac/create.go

```diff
@@ -114,6 +114,43 @@ func CreateGlobalRoleBinding(client *rancher.Client, globalRoleBinding *v3.Globa
 	return newGlobalRoleBinding, nil
 }
 
+// CreateRoleTemplate is a helper function that uses the dynamic client to create a cluster/project role template for a specific user
+func CreateRoleTemplate(client *rancher.Client, roleTemplateName, roleContext string, roleTemplateIDs []string, annotations, labels map[string]string, rules []rbacv1.PolicyRule, builtin, locked bool) (*v3.RoleTemplate, error) {
```

Comment:
Can we pass the roletemplate as input to this function instead of having so many input parameters, similar to what we've done for global role [here](https://github.com/rancher/shepherd/blob/80035045f223836a0b072348c378c7a58be885f4/extensions/kubeapi/rbac/create.go#L74)? This approach would make it easier to integrate any new fields introduced in the role template without having to modify this function.